packages/@react-spectrum/s2/src/Accordion.tsx

```diff
@@ -0,0 +1,397 @@
+    }
+  },
+  height: 'full',
+  minHeight: 'control',
+  width: 'full',
+  backgroundColor: {
+    default: 'transparent',
+    isFocusVisible: 'transparent-black-100',
+    isHovered: 'transparent-black-100'
+  },
+  borderWidth: 0,
+  borderRadius: {
+    // Only rounded for keyboard focus and quiet hover.
+    default: 'none',
+    size: {
+      S: {
+        isFocusVisible: '[7px]',
+        isQuiet: {
+          isHovered: '[7px]',
+          isFocusVisible: '[7px]'
+        }
+      },
+      M: {
+        isFocusVisible: 'default',
+        isQuiet: {
+          isHovered: 'default',
+          isFocusVisible: 'default'
+        }
+      },
+      L: {
+        isFocusVisible: '[9px]',
+        isQuiet: {
+          isHovered: '[9px]',
+          isFocusVisible: '[9px]'
+        }
+      },
+      XL: {
+        isFocusVisible: 'lg',
+        isQuiet: {
+          isHovered: 'lg',
+          isFocusVisible: 'lg'
+        }
+      }
+    }
+  },
+  textAlign: 'start',
+  disableTapHighlight: true
+});
+
+const chevronStyles = style({
+  rotate: {
+    isRTL: 180,
+    isExpanded: 90
+  },
+  transitionDuration: '100ms',
+  transitionProperty: 'rotate',
+  '--iconPrimary': {
+    type: 'fill',
+    value: 'currentColor'
+  },
+  flexShrink: 0
+});
+
+function AccordionHeader(props: AccordionHeaderProps, ref: DOMRef<HTMLDivElement>) {
+  let {
+    level = 3,
+    UNSAFE_style,
+    UNSAFE_className = '',
+    ...otherProps
+  } = props;
+  let domRef = useDOMRef(ref);
+  const domProps = filterDOMProps(otherProps);
+  let {direction} = useLocale();
+  let {isExpanded} = useContext(DisclosureStateContext)!;
+  let {size, density, isQuiet} = useContext(InternalAccordionItemContext);
+  let isRTL = direction === 'rtl';
+  return (
+    <Heading
+      {...domProps}
+      level={level}
+      ref={domRef}
+      style={UNSAFE_style}
+      className={(UNSAFE_className ?? '') + headingStyle}>
+      <Button className={({isHovered, isFocused, isFocusVisible, isDisabled}) => buttonStyles({size, isHovered, isFocused, isFocusVisible, density, isQuiet, isDisabled})} slot="trigger">
+        <Chevron size={size} className={chevronStyles({isExpanded, isRTL})} aria-hidden="true" />
+        {props.children}
+      </Button>
+    </Heading>
+  );
+}
+
+/**
+ * A header for an accordion item. Contains a heading and a trigger button to expand/collapse the panel.
+ */
+let _AccordionHeader = /*#__PURE__*/ (forwardRef as forwardRefType)(AccordionHeader);
+export {_AccordionHeader as AccordionHeader};
+
+export interface AccordionPanelProps extends RACAccordionPanelProps, UnsafeStyles, DOMProps, AriaLabelingProps {
+  children: React.ReactNode
+}
+
+const panelStyles = style({
+  font: 'body',
+  paddingTop: {
+    isExpanded: sizeValue(8)
+  },
+  paddingBottom: {
+    isExpanded: sizeValue(16)
+  },
+  paddingX: {
+    isExpanded: {
+      size: {
+        S: sizeValue(8),
+        M: sizeValue(9),
+        L: sizeValue(12),
+        XL: sizeValue(15)
```

Comment:
hmm wonder where these sizes came from. can they be derived from something? Probably to align with the chevron icon?